render/engine.go

```diff
@@ -0,0 +1,115 @@
+// Render shouldn't have to know about the implementations of it's base render package. It should simply
```

Comment:
This is good, it makes no reference to .yml so in theory (with enough updates) we can move to support JSON spec files or any other types in the future.

Reply:
The tricky part of this is unmarshalling into the composed struct type. If some information needs to live within the Node portion of any given struct, we need to get it there through the use of a helper method.